### ISSUE TITLE
[FEATURE REQ] Export outputs to JSON

### ISSUE THREAD

**Issue by Samyak2** [pesos/browser-history#35]
**Is your feature request related to a problem? Please describe.**

Currently, the outputs are a tuple of urls and timestamps. It would be useful to have an option to export this data to a readable file such as JSON.

**Describe the solution you'd like**

A function in `generic.Outputs` called `to_json` that exports the history to the given filename in JSON format.

The exact structure of the JSON is discussed in [this comment](https://github.com/pesos/browser-history/issues/4#issuecomment-689432550)

The `to_json` function should, by default, export in the first format and a parameter must be provided to get output in the 2nd format (refer the comment linked above)

**Describe alternatives you've considered**

Other issues are opened for alternate formats (if you can't find an issue for the format you need, please open one)

**Additional context**

#4

**Comment by lucasace** [comment]
Only the function to be created right , does it need to create the cli as well?

**Comment by lucasace** [comment]
I would like to do this as well

**Comment by Samyak2** [comment]
Hi. Since this is a beginner friendly repo, I want to keep as many unassigned issues open as possible (to make it easier for people to contribute) and since you're already assigned to another issue, I'll keep this issue open for others if they're interested. You're welcome to do this once your other PR is merged.

**Comment by SanchitJain123** [comment]
Hi @Samyak2. Can I work on this, I'll open PR during Hacktoberfest.

**Comment by Samyak2** [comment]
Sure

**Comment by SanchitJain123** [comment]
@Samyak2 How can I access the metadata from `generic.Outputs` instance that you have mentioned in the comment, sorry but I was unable to find the metadata fields required in the plain JSON format from the same.

**Comment by sujanchegu** [comment]
Hey @SanchitJain123 , So by `metadata` he means the basic properties of an `Outputs` instance such as the name of the browser and the like. I've opened an issue now #49 .  I feel that issue should be solved after solving this issue. If you want to work on that issue, you can comment on that issue and get it assigned to you. However, if you want to continue with this, go right ahead. Any help you need, ask us here or on our slack channel. :smile:

**Comment by SanchitJain123** [comment]
Thanks for the information @sujanchegu. I have added a function named to_json referred by @Samyak2. I feel the only thing missing is `metadata` for the Plain JSON format. So, should I open a pull request for this or is it mandatory to add metadata in the Plain JSON format for this issue?

**Comment by sujanchegu** [comment]
@SanchitJain123, I would say open the PR.  Let the assignee who works on that issue manage the metadata part. That way, it's elegant and would mean that issues are solved in their corresponding PRs.

**Comment by SanchitJain123** [comment]
Thank you @sujanchegu for helping.

**Comment by HardipinderS** [comment]
is a branch created for this ?

**Comment by sujanchegu** [comment]
No separate branch is created for this. However, this issue is already assigned to somebody else. You can review the above PR with your inputs/ideas or take up any other issue. :smile:
